packages/userstorage/src/browser/user-storage-service-filesystem.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { Disposable, DisposableCollection, ILogger, Emitter, Event } from '@theia/core/lib/common';
+import { UserStorageChangeEvent, UserStorageService } from './user-storage-protocol';
+import { injectable, inject } from 'inversify';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import { FileSystemWatcherServer, DidFilesChangedParams } from '@theia/filesystem/lib/common/filesystem-watcher-protocol';
+import * as jsoncparser from 'jsonc-parser';
+import URI from '@theia/core/lib/common/uri';
+import { UserStorageUri } from './user-storage-uri';
+
+@injectable()
+export class UserStorageServiceFilesystem implements UserStorageService {
+
+    protected readonly toDispose = new DisposableCollection();
+    protected readonly onUserStorageChangedEmitter = new Emitter<UserStorageChangeEvent>();
+    protected rootURI: URI;
+
+    constructor(
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(FileSystemWatcherServer) protected readonly watcherServer: FileSystemWatcherServer,
```

Comment:
in the frontend `FileSystemWatcher` should be used, `FileSystemWatcherServer` it is the implementation details of `FileSystemWatcher`